### ISSUE TITLE
help for dropdown menu

### ISSUE THREAD

**Issue by marieeved** [wet-boew/wet-boew#6954]
hi,
 in the WET, we don't have the code for a dropdown menu. I already read messages on git hub about that, that you will not add such a code... but canada.ca have a dropdown menu. They use an other technique : "data-ajax". I don't know this tag. I tried to use it, I change the path, I made few modifications to the doc : sitemenu-fr.html (with my own labels and links) and it didn't work. Can someone help me with this?

**Comment by LaurentGoderre** [comment]
the data-ajax-replace is a custom attribute for WET that lets load the full menu via AJAX. The first thing to make sure when trying it is that you run it on a server. If you are running it on your computer without a server, the browser will block the ajax request.

**Comment by LaurentGoderre** [comment]
P.S. the dropdowns that we are not bringing are those ones: http://getbootstrap.com/components/#btn-dropdowns

**Comment by pjackson28** [comment]
@marieeved If you are referring to the menu bar then have you seen the working example (http://wet-boew.github.io/wet-boew/demos/menu/menu-en.html) and the documentation (http://wet-boew.github.io/wet-boew/docs/ref/menu/menu-en.html)?

**Comment by marieeved** [comment]
thanks Laurent. it means that I'm not so bad... effectively, I was running it just on my computer and not on the server. :)

**Comment by marieeved** [comment]
@pjackson28 I already seen that, but I will check again. Maybe I made a mistake. Thanks.

**Comment by LaurentGoderre** [comment]
Let me know if you want suggestions for a we server you could run locally.

**Comment by marieeved** [comment]
I made a menu dropdown. :)  but the menu is not horizontal but vertical.

**Comment by marieeved** [comment]
ok even with ajax and with a server, it doesn't work! so something is really poorly explained in the wet... because I don't understand how it can work.

**Comment by LaurentGoderre** [comment]
Can you take a screenshot of the network tab in your developers tools and add it here?

**Comment by marieeved** [comment]
![dropdown](https://cloud.githubusercontent.com/assets/13418868/8802645/c562c656-2f90-11e5-98eb-f6a497effbce.JPG)

**Comment by marieeved** [comment]
this code works but the dropdown is vertical because it's the sub sub menu. but I'm not able to create the dropdown for the sub menu.

**Comment by LaurentGoderre** [comment]
It's probably best to stick with the current format for the menu

https://github.com/wet-boew/wet-boew-dist/blob/v4.0-dist/unmin/ajax/sitemenu-en.html

**Comment by marieeved** [comment]
ok so I have to absolutly use the ajax technique? with a separate file.

**Comment by LaurentGoderre** [comment]
Is there any reason why you wouldn't want to use the ajax approach?

**Comment by marieeved** [comment]
well, because it doesn't work. I already tried. problem with the anchors maybe?

**Comment by marieeved** [comment]
![html](https://cloud.githubusercontent.com/assets/13418868/8803199/9138c5d0-2f93-11e5-9991-92ce84e32ac6.JPG)
this is my code in my page html named "test".

**Comment by LaurentGoderre** [comment]
I'm confused, you said it worked earlier, so it doesn't work? It doesn't load?

**Comment by LaurentGoderre** [comment]
This must be an older version of the code. `data-ajax-fetch` is deprecated in favor of `data-ajax-replace` now. It shoudl still work though. Can you take a screenshot of your network tab in your developers tools

**Comment by marieeved** [comment]
it works without ajax. but the dropdown is inversed.
instead to have: section 1 section 2 section 3 in the same line (like this).
I have a dropdown like this:
section 1
section 2
 section 3

**Comment by LaurentGoderre** [comment]
Can you add a screenshot please?

**Comment by marieeved** [comment]
this is the code of the dropdown that works but inversed:
![dropdown](https://cloud.githubusercontent.com/assets/13418868/8803347/6baa7498-2f94-11e5-9ee0-f2c0347ae6fe.JPG)

**Comment by LaurentGoderre** [comment]
There is no details sumamry in your top menu, you should follow this syntax for your menu

https://github.com/wet-boew/wet-boew-dist/blob/v4.0-dist/unmin/ajax/sitemenu-en.html

**Comment by marieeved** [comment]
I know... thanks, but I will come back later. I have to do something else.

**Comment by marieeved** [comment]
thanks a lot. if you tell me that it supposed to work with ajax. then it's probably an error of path or maybe anchor... I will try again.

**Comment by marieeved** [comment]
http://wet-boew.github.io/wet-boew/demos/menu/menu-fr.html
I copy/paste the code of this page (I just change the path for have the right file:sitemenu-fr) with a separate file for ajax. I upload the 2 file with the server.
and it still doesn't work.

**Comment by LaurentGoderre** [comment]
The first step is to make sure that the ajax request was sent. Please add a screenshot of your network tab in your dev tools.

**Comment by marieeved** [comment]
![html2](https://cloud.githubusercontent.com/assets/13418868/8805140/ad7d079c-2f9d-11e5-87c8-3f110636db40.JPG)

![ajax2](https://cloud.githubusercontent.com/assets/13418868/8805141/b095eab6-2f9d-11e5-9d5b-797bbcac347b.JPG)

**Comment by LaurentGoderre** [comment]
I mean, a screenshot of your browser's developer tool that show the list of HTTP request for the page (generally you can get to it by using F12 and then Network tab)

**Comment by marieeved** [comment]
F12 on the page that I test, right? ok for network, there is nothing. it's white.
![net](https://cloud.githubusercontent.com/assets/13418868/8805275/56618cc0-2f9e-11e5-9674-e8c10fd6bd2b.JPG)

**Comment by LaurentGoderre** [comment]
Do you have access to anything else than IE? If not, you need to hit the run button (Green arrow in the top left corner) then reload the page. This is only in IE, other browsers display it automatically.

**Comment by marieeved** [comment]
when you said reload the page, you mean republish it with Dreamweaver?

**Comment by marieeved** [comment]
![network](https://cloud.githubusercontent.com/assets/13418868/8805753/f8c0013e-2fa0-11e5-88bb-9336ff3a0257.JPG)

**Comment by LaurentGoderre** [comment]
Ok, as you can see, the third one from the bottom load without any problem. Is there any javascript error in the console (Console tab)

**Comment by marieeved** [comment]
yes I have an error since the last few days and I'm not able to change anything. it affects other things like I can't see tabs on a page when I load only local (and I was able to see it before). So how can I debug a page?

**Comment by marieeved** [comment]
and yes I have a script in the page to display auto the date of last modification. 
when I open one of my page I received an error message close to:"impossible to obtain ownership of a non-zero href or undefined". I searched on internet how to debug a page but it's not really clear how to do it.

**Comment by LaurentGoderre** [comment]
wow....that is obscure! I have never seen ti before.

**Comment by marieeved** [comment]
I will ask on forum "comment ça marche" if someone can help me. Thanks for everything @LaurentGoderre. like you said if I have an error maybe that's why the dropdown doesn't work. I hope so. :)

**Comment by LaurentGoderre** [comment]
Can you add a screenshot of the console error?

**Comment by marieeved** [comment]
![error-console](https://cloud.githubusercontent.com/assets/13418868/8835165/b9e8ff92-3086-11e5-9768-aa26b3460e46.JPG)

**Comment by LaurentGoderre** [comment]
This we can work with! Can you temporarily replace all instances of `.min.js` in your HTML file to switch to the unminified files (makes it easier to debug) and post a new screenshot of the error?

**Comment by marieeved** [comment]
I'm not sure of what you mean... you want that I erase all reference to min.js in my html code or you want that I replace it by something else?

**Comment by LaurentGoderre** [comment]
Ooops, I meant replace `.min.js` with `.js`

**Comment by marieeved** [comment]
it only written .min.js in a comment. I think that the page don't accept script because it written no script everywhere.
![code-script](https://cloud.githubusercontent.com/assets/13418868/8836317/1aafb184-308e-11e5-9af3-b2ebb18b50d0.JPG)

**Comment by LaurentGoderre** [comment]
No, these are old IE scripts. the ones your are looking for are at the bottom of the page (last elements in the body).

**Comment by marieeved** [comment]
oh I see... it's done. the error disappear.it's normal?
![code2](https://cloud.githubusercontent.com/assets/13418868/8836467/33509108-308f-11e5-9638-8f6748fb7b50.JPG)

**Comment by marieeved** [comment]
when I load it on the server... I got this:
![code-server](https://cloud.githubusercontent.com/assets/13418868/8836551/ae6e240e-308f-11e5-83e6-e25c56d5aae0.JPG)

**Comment by marieeved** [comment]
@LaurentGoderre : I must leave office. Again, thank a lot for your time. I will read your comments tomorrow morning.

**Comment by marieeved** [comment]
maybe an other thing to think... my boss just show me this: http://wet-boew.github.io/wet-boew/docs/server-en.html 
and we didn't do that. origin of my problem? and if so, I will need help understand the instructions. :)

**Comment by LaurentGoderre** [comment]
Not doing this would not cause a JavaScript or an error. Can you make sure that the unminified javascript load properly (screenshot of the network tab)

**Comment by marieeved** [comment]
I load the page on the server. than F12 display this:
![network2](https://cloud.githubusercontent.com/assets/13418868/8850516/a7e059a4-3116-11e5-8c5a-74a2f4da8322.JPG)

**Comment by LaurentGoderre** [comment]
Why are all the references using the file:// protocol? Something is not right

**Comment by LaurentGoderre** [comment]
Ooooh, it was there from the start...well that is your problem. You can't use the file protocol, the browser blocks those. You need to put those assets on the server as well and fix your paths.

**Comment by marieeved** [comment]
if it's just that, that would be great! I confess I had not checked ... I work in a local folder and I created a folder  only a few days ago to test with the server. I have to publish a press review, but I look at it right after. thanks. :)

**Comment by LaurentGoderre** [comment]
There are options to run servers on your machine to make it easier for you to test.

**Comment by marieeved** [comment]
yes you can tell me about it.

**Comment by marieeved** [comment]
I loaded all the files on the server (pic, css, ajax, file html). and when I press F12, I got this:
![net3](https://cloud.githubusercontent.com/assets/13418868/8854016/a804c40a-312b-11e5-8488-aa0586753681.JPG)
![script3](https://cloud.githubusercontent.com/assets/13418868/8854019/ac5b8f8e-312b-11e5-88aa-93518aaf73fb.JPG)
![error3](https://cloud.githubusercontent.com/assets/13418868/8854022/b04bb0f6-312b-11e5-9771-11db3459db51.JPG)

**Comment by LaurentGoderre** [comment]
Much better, again can you switch to using the unminified file? If you are using version 4.0.15 or later it would be `intranet/js/wet-boew.js` in earlier versions of WET it would be `intranet/unmin/js/wet-boew.js`

**Comment by marieeved** [comment]
ah ok!. yes give me 10 min...

**Comment by marieeved** [comment]
I used the wet 4.0.12 so you want that change what i have for this path:  intranet/unmin/js/wet-boew.js right?
now, I try to redownload the wet...I think we don't have anymore this folder "unmin", maybe deleted by error. i'm not the only one on the project.

**Comment by marieeved** [comment]
effectively 2 folders missing... I change the path to use "unmin". I load all the files on the server. and I get something different with the dev tool.
![net4](https://cloud.githubusercontent.com/assets/13418868/8855307/85f04bb2-3132-11e5-9e79-5a2b71ad8f74.JPG)
![error4](https://cloud.githubusercontent.com/assets/13418868/8855312/8c8a9658-3132-11e5-93fc-d4388f799165.JPG)
![script4](https://cloud.githubusercontent.com/assets/13418868/8855318/91d11e52-3132-11e5-913d-37f15c900289.JPG)

**Comment by LaurentGoderre** [comment]
That error is caused by the syntax of your menu I believe. Are you following the syntax here:

http://wet-boew.github.io/wet-boew/demos/menu/menu-fr.html

**Comment by marieeved** [comment]
yes. I follow this code.
one thing is the anchor: #lore and id=lore... it coulb de anything? because I have more than 3 sections in my menu.

**Comment by LaurentGoderre** [comment]
yeah, as long as the pair matches. Can you put your code here?

**Comment by marieeved** [comment]
![html1](https://cloud.githubusercontent.com/assets/13418868/8856965/30aabfd4-313c-11e5-8808-e15be063fc4b.JPG)

**Comment by marieeved** [comment]
![html1-1](https://cloud.githubusercontent.com/assets/13418868/8856994/65a0622a-313c-11e5-9e66-30a04a29993e.JPG)
lign 38: `</html>`

**Comment by marieeved** [comment]
everything seem the same. I checked. I just changed color of the menu.

**Comment by LaurentGoderre** [comment]
I don't think I can help more without seeing it but the format you are using is not the same, or perhaps you are not loading the file you think you are loading. As you can see from your screenshot from the debug mode, the section is a heading element while it should be an `a` element.

**Comment by LaurentGoderre** [comment]
Unless!!!! Do you have a side menu?

**Comment by marieeved** [comment]
no we don't want side menu.it's more interresting to have a dropdown. everyone have one.

i'm not sure I understand your last comment. how the format could be different if I have copy/paste the code from the page you refered me all the time.
the code html is inside the header element because that's what it is written on : http://wet-boew.github.io/wet-boew/docs/ref/menu/menu-en.html 
if it's not what I must do, can you just tell me what i have to change.
I used the same code that it given on the web pages of the wet.

Is there a place (on the web pages of the wet or on the git hub) where the ajax technology and drop-down menus are explained more in detail?? I could handle it by myself.

**Comment by LaurentGoderre** [comment]
The error is not related to ajax because it did it's job and loaded the html, however WET is not able to process it. Is there any way you can put the code in a github repo or send it to me via email?

**Comment by marieeved** [comment]
send you by email. thanks. come back in 45 min.

**Comment by marieeved** [comment]
come back tomorrow. thanks again.

**Comment by LaurentGoderre** [comment]
The problem actually has nothing to do with the menu but with the fact that you heavily diverged from the WET codebase. Did you start with the WET theme and try to create a new one out of it?

**Comment by LaurentGoderre** [comment]
The error is caused by the footer markup which doesn't respect the codebase. If you remove it, the menu load. However there is still an error and the mobile menu fails to load.

I would highly encourage to stick to the existing themes as these heavy customization will break WET and make it difficult for mobile users to use the site.

**Comment by LaurentGoderre** [comment]
Actually, without the footer there are no more error but the mobile view and mobile menu is really messed up.

**Comment by marieeved** [comment]
I think my colleague before me done this:
-  download :themes-dist-4.0.12-theme-gc-intranet
- then using code of a page from an other gc intranet page and then change the code  in using some of elements available in the wet (secondary menu, all the styles with the css files).

so If I understand it was wrong to do this. we miss some steps? on this page: http://wet-boew.github.io/wet-boew/docs/start-en.html
I must to follow steps of option 2: want to use wet but design new one??
or I have to use a page of content in intranet theme of wet and then change the nav, content and footer with my labels and links, etc. ?

**Comment by LaurentGoderre** [comment]
Is this for an intranet site?

**Comment by marieeved** [comment]
yes.

**Comment by LaurentGoderre** [comment]
I would recommend sticking to the already established intranet theme. It is possible to create a theme but it requires a deep understanding of WET and would leave you to support it which can be a lot of work.

**Comment by marieeved** [comment]
ok so step by step. I must do what exactly?
I click F12, copy/paste code of this page: http://wet-boew.github.io/themes-dist/theme-gc-intranet/content-fr.html
and change content but not structure of the page to create my pages, that's it?

but I do what to create an home page because there is no model for the intranet theme : http://wet-boew.github.io/themes-dist/theme-gc-intranet/index-en.html#fr

**Comment by marieeved** [comment]
well, I take the content page for the home page, right?

and the content page-secondary menu for the other pages, right? (if I want a secondary menu... if not I only take the content page to create my pages?)

**Comment by marieeved** [comment]
oups, I don't need to copy paste the code. we already have the page in the folder. so I change it directly when I opened it with Dreamweaver, right?

**Comment by LaurentGoderre** [comment]
Sounds right yes

**Comment by marieeved** [comment]
I make a try. the dropdown works!!!!!!!!!!!!!!!!!!!!!!
:) :) :)
thanks a lot @LaurentGoderre

**Comment by marieeved** [comment]
last questions:
but I still can modify (only element not structure) by ex: the search box...I don't want that it search in Google but in one of our page. it's ok?

and can I use a script to display last modification auto instead of you code because I have to change it manually?

**Comment by marieeved** [comment]
now i try to change the ajax-menu to add some labels and even in uploading all the files on the server, now it doesn't make the change.... :(

**Comment by pjackson28** [comment]
Note that the search was just an example implementation. You would configure it to point to whatever search engine you need (although keeping the rest of the search markup the same).

Also note that the last modified date should reflect the currency of the content so should not be automated unless you have a means to display the actual date of the last significant modification to the content (which does not include header, footer, etc).

Also you are not seeing the change because the menu is being cached. Clear the cache, or change your cache settings to check each time for an update if you want to see the update. IE seems to by default not be configured to check each time for an update.

**Comment by marieeved** [comment]
again, thanks a lot.
I'll close this one. if I have other questions, I will open a new issue.
